spec/minimax_rb_spec.rb

```diff
@@ -1,145 +1,217 @@
-    expected = [[0, 1, 2], [3, 4, 5], [6, 7, 8]]
-
-    expect(MinimaxRB.rows(board)).to eq expected
-  end
-
-  it "returns the columns of a board" do
-    board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
-    expected = [[0, 3, 6], [1, 4, 7], [2, 5, 8]]
-
-    expect(MinimaxRB.columns(board)).to eq expected
-  end
-
-  it "returns the diagonals of a board" do
-    board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
-    expected = [[0, 4, 8], [6, 4, 2]]
-
-    expect(MinimaxRB.diagonals(board)).to eq expected
-  end
-
-  it "with the highest score/move pair, returns the move" do
-    expect(MinimaxRB.max_move({5 => 1})).to eq 5
-  end
+      expect(MinimaxRB.columns(board)).to eq expected
+    end
 
-  it "with the highest score/move pair, returns the score" do
-    expect(MinimaxRB.max_score({5 => 1})).to eq 1
+    it "returns the diagonals of a board" do
+      board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+      expected = [[0, 4, 8], [6, 4, 2]]
+
+      expect(MinimaxRB.diagonals(board)).to eq expected
+    end
+  end
+
+  context "game rules" do
+    it "checks if a winning combo contains a specific marker, returns true" do
+      marker_o = "O"
+      o_winning_combo = ["O", "O", "O"]
+
+      expect(MinimaxRB.marker_winning_combo?(marker_o, o_winning_combo)).to eq true
+    end
+
+    it "checks if a winning combo contains a specific marker, returns false" do
+      marker_o = "O"
+      o_winning_combo = ["O", "O", "X"]
+
+      expect(MinimaxRB.marker_winning_combo?(marker_o, o_winning_combo)).to eq false
+    end
+
+    it "returns true if a specific marker won" do
+      marker_x = "X"
+      x_winning_board = ["X", nil, nil, "X", nil, nil, "X", nil, nil]
+
+      expect(MinimaxRB.marker_winner?(marker_x, x_winning_board)).to eq true
+    end
+
+    it "returns false if no specific marker won" do
+      marker_x = "X"
+      x_winning_board = ["X", nil, nil, "X", nil, nil, nil, nil, nil]
+
+      expect(MinimaxRB.marker_winner?(marker_x, x_winning_board)).to eq false
+    end
+    it "returns true for a draw" do
+      board = ["X", "O", "X", "O", "X", "O", "O", "X", "O"]
+      marker = "X"
+      opponent = "O"
+      empty_marker = nil
+
+      expect(MinimaxRB.draw?(board, marker, opponent, empty_marker)).to eq true
+    end
+
+    it "returns false for not a draw" do
+      board = ["X", "O", "X", "O", "X", "O", "O", "X", nil]
+      marker = "X"
+      opponent = "O"
+      empty_marker = nil
+
+      expect(MinimaxRB.draw?(board, marker, opponent, empty_marker)).to eq false
+    end
+
+    it "returns true for a win" do
+      board = ["X", "X", "X", "O", "O", nil, nil, nil, nil]
+      marker = "X"
+      opponent = "O"
+
+      expect(MinimaxRB.win?(board, marker, opponent)).to eq true
+    end
+
+    it "returns false for not a win" do
+      board = ["X", "X", nil, "O", "O", nil, nil, nil, nil]
+      marker = "X"
+      opponent = "O"
+
+      expect(MinimaxRB.win?(board, marker, opponent)).to eq false
+    end
+  end
+
+  context "best move" do
```

Comment:
`context` and `describe` both do the same thing (allow you to group related specs), but `describe` might be more idiomatic for RSpec in this case.  `context` is usually when some sort of input data or environment dictate special-case behavior.

Not a big deal; just something to be aware of.